core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -543,4 +512,181 @@ public TenantObject setDefaults(final JsonObject defaultProperties) {
         setProperty(TenantConstants.FIELD_PAYLOAD_DEFAULTS, Objects.requireNonNull(defaultProperties));
         return this;
     }
+
+    /**
+     * Sets the configuration information for this tenant's configured trusted certificate(s).
+     * 
+     * @param configuration A single JSON object property or a list of JSON objects, one set
+     *            of properties for each configured trusted CA.
+     * @return This tenant for command chaining.
+     * 
+     * @throws IllegalArgumentException if the any of the trusted CA configuration is missing the required
+     *             <em>subject-dn</em> property or either the {@code public-key} or {@code cert} property does not
+     *             contain a valid Base64 encoding.
+     */
+    @SuppressWarnings("unchecked")
+    @JsonProperty(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA)
+    public TenantObject setTrustedCa(final Object configuration) {
```

Comment:
I have to admit that I do not really like the `Object` type parameter. In which case is this method invoked? Can we maybe replace it with two variants, one accepting a `JsonArray`, the other a `JsonObject`?

Reply:
> In which case is this method invoked?

this method is invoked by the `Jackson` library when mapping a `JsonObject` to a `TenantObject` i.e during a call to `JsonObject.mapTo(TenantObject.class)`

> Can we maybe replace it with two variants, one accepting a JsonArray, the other a JsonObject?

I actually tried this first, however, If I should separate it into two methods, then I cannot annotate both methods using the same property, i.e `trusted-ca`. If the annotation is missing, then the method will not be called during the mapping process. 

This is why I decided to put them in a single method and annotate the single method.